qiskit_dynamics/signals/signals.py

```diff
@@ -441,24 +415,24 @@ def add_samples(self, start_sample: int, samples: List):
         Raises:
             QiskitError: If start_sample is less than the current length of samples.
         """
-        samples = Array(samples)
+        samples = unp.asarray(samples)
 
         if len(samples) < 1:
             return
 
         if start_sample < len(self.samples):
             raise QiskitError("Samples can only be added afer the last sample.")
 
-        zero_pad = np.expand_dims(np.zeros_like(Array(samples[0])), 0)
+        zero_pad = np.expand_dims(np.zeros_like(unp.asarray(samples[0])), 0)
```

Comment:
I think like the above case these `np`s should be changed to `unp`s. I think this is fine for JAX, but I wonder if it would pose problems for tensorflow?

Reply:
modified at https://github.com/Qiskit/qiskit-dynamics/pull/208/commits/10742b902c76863ba805c983c4dca91db22afd81.